File: problems/tier2-intermediate/013-parking-lot/boilerplate/java/part2/Guided.java
import java.util.*;

// Data class (given).
class ParkOp {
    public String kind;
    public String s1;
    public String s2;
    public String s3;
    public int i1;
    public int i2;
    public int i3;

    public ParkOp(String kind, String s1, String s2, String s3, int i1, int i2, int i3) {
        this.kind = kind;
        this.s1 = s1;
        this.s2 = s2;
        this.s3 = s3;
        this.i1 = i1;
        this.i2 = i2;
        this.i3 = i3;
    }

    public ParkOp(String kind) {
        this(kind, "", "", "", 0, 0, 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> parking_simulate(List<ParkOp> ops) {
        // TODO: write your solution
        return null;
    }

}

File: problems/tier1-foundation/006-discount-engine/boilerplate/cpp/part1/guided.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;


// Data class (given).
struct CartItem {
    string name;
    double price;
    int quantity;
    string category;
    CartItem(const string& name_, double price_, int quantity_, const string& category_ = "")
      : name(name_), price(price_), quantity(quantity_), category(category_) {}
};

// HINT: introduce an abstraction so new variants don't change existing code.
// HINT: keep each piece small — one responsibility per class.

// HINT: start from what this must return, then work backwards to the state it needs.
double apply_percentage_discount(vector<CartItem> cart, double percentage) {
    // TODO: write your solution
    return {};
}

// HINT: start from what this must return, then work backwards to the state it needs.
double apply_flat_discount(vector<CartItem> cart, double amount) {
    // TODO: write your solution
    return {};
}

// HINT: start from what this must return, then work backwards to the state it needs.
double apply_bogo(vector<CartItem> cart, int buyCount, int freeCount) {
    // TODO: write your solution
    return {};
}
